### PR TITLE
Remove star imports from diffeq and filtsmooth (and filtsmooth.statespace)

### PR DESCRIPTION
**Pull request by pnkraemer**
The name says it all.

No more star imports in `__init__.py` files (in diffeq and filtsmooth). This avoids weird exports of `statespace.numpy` and the like.

### CODE REVIEW

File: src/probnum/diffeq/ode/__init__.py [hunk 1]
```diff
@@ -1,6 +1,14 @@
-from .ivp import *
-from .ivp_examples import *
-from .ode import *
+from .ivp import IVP
+from .ivp_examples import (
```
Comment (schmidtjonathan) [inline src/probnum/diffeq/ode/__init__.py:2]: You do not import the `*_rhs()` methods from `ivp_examples`. I personally find that desirable, I just wanted to check whether that was a deliberate choice.

File: src/probnum/filtsmooth/__init__.py [hunk 2]
```diff
@@ -1,8 +1,19 @@
...
+    ContinuousUKFComponent,
+    DiscreteEKFComponent,
+    DiscreteUKFComponent,
+    FixedPointStopping,
+    IteratedKalman,
+    Kalman,
+    KalmanPosterior,
+    StoppingCriterion,
+    UnscentedTransform,
+)
```
Comment (schmidtjonathan) [inline src/probnum/filtsmooth/__init__.py:17]: Shouldn't we also import the `statespace` module here?

File: src/probnum/filtsmooth/statespace/sde.py [hunk 3]
```diff
@@ -29,6 +29,7 @@ def __init__(
         self.dispmatfun = dispmatfun
         self.jacobfun = jacobfun
         self.precon = None
```
Comment (schmidtjonathan) [inline src/probnum/filtsmooth/statespace/sde.py:31]: `self.precon = None` is already set in `Transition.__init__()`. I suggest removing this line and moving the `super().__init__()` call to the top of `SDE`'s initializer.

File: src/probnum/filtsmooth/__init__.py [hunk 4]
```diff
@@ -1,8 +1,19 @@
 """Bayesian Filtering and Smoothing."""
 
-from .bayesfiltsmooth import *
-from .filtsmoothposterior import *
-from .gaussfiltsmooth import *
+from .bayesfiltsmooth import BayesFiltSmooth
+from .filtsmoothposterior import FiltSmoothPosterior
+from .gaussfiltsmooth import (
```
Comment (schmidtjonathan) [inline src/probnum/filtsmooth/__init__.py:5]: Actually, I think you forgot to replace the `*`-imports in `filtsmooth.gaussfiltsmooth.__init__.py`

**Review by schmidtjonathan — CHANGES_REQUESTED** [review]
That's a lot more clean and concise, thanks for taking care of this! I added a few comments, most of which are minor. The imports from `filtsmooth.gausfiltsmooth.__init__.py` are probably most important.

File: src/probnum/diffeq/ode/__init__.py (hunk 1 above)
Comment (pnkraemer) [reply src/probnum/diffeq/ode/__init__.py:2]: This was a deliberate choice

File: src/probnum/filtsmooth/__init__.py (hunk 2 above)
Comment (pnkraemer) [reply src/probnum/filtsmooth/__init__.py:17]: is this how it works? I thought having it separate would be sufficient...

File: src/probnum/filtsmooth/__init__.py (hunk 4 above)
Comment (pnkraemer) [reply src/probnum/filtsmooth/__init__.py:5]: hmmmmm seems it. I thought I had... thanks for the pointer!

**Review by schmidtjonathan — APPROVED** [review]
:+1:

File: src/probnum/filtsmooth/__init__.py (hunk 2 above)
Comment (schmidtjonathan) [reply src/probnum/filtsmooth/__init__.py:17]: Before merging, you might want to reconsider that. I think, it works as it is, but I see no reason why in the `filtsmooth.__init__.py` we should explicitly import `gaussfiltsmooth` but not `statespace`.

**Comment by schmidtjonathan** [comment]
I added one more comment you might want to consider [here](https://github.com/probabilistic-numerics/probnum/pull/284#discussion_r543533797)

**Pull request merged**